packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,276 @@
+import { injectIntl } from 'react-intl'
+import { linkState } from 'reaclette-utils'
+import { Container, Row, Col } from 'grid'
+import { provideState, injectState } from 'reaclette'
+import { SelectPool, SelectPif } from 'select-objects'
+import { addSubscriptions, connectStore } from 'utils'
+import {
+  getBondModes,
+  createBondedNetwork,
+  createNetwork,
+  subscribeCurrentUser,
+} from 'xo'
+import {
+  createSelector,
+  createGetObject,
+  getIsPoolAdmin,
+  getUser,
+} from 'selectors'
+
+import styles from './index.css'
+import Page from '../../page'
+
+const getInitialState = () => ({
+  bondedNetwork: false,
+  bondMode: undefined,
+  description: '',
+  mtu: '',
+  name: '',
+  pif: undefined,
+  pifs: undefined,
+  vlan: '',
+})
+
+const NewNetwork = decorate([
+  addSubscriptions({
+    user: subscribeCurrentUser,
+  }),
+  connectStore(() => ({
+    isAdmin: createSelector(
+      getUser,
+      user => user && user.permission === 'admin'
+    ),
+    isPoolAdmin: getIsPoolAdmin,
+    pool: createGetObject((_, props) => props.location.query.pool),
+  })),
+  injectIntl,
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      linkState,
+      onChangeBondedNetwork: () => ({ bondedNetwork }) => ({
+        ...getInitialState(),
+        bondedNetwork: !bondedNetwork,
+      }),
+      onChangePif: (_, value) => ({ bondedNetwork }) =>
+        bondedNetwork ? { pifs: value } : { pif: value },
+      reset: getInitialState,
+    },
+    computed: {
+      pifPredicate: (_, { pool }) => pif =>
+        pif.vlan === -1 && pif.$host === (pool && pool.master),
+    },
+  }),
+  injectState,
+  class extends Component {
+    static contextTypes = {
+      router: PropTypes.object,
+    }
+
+    componentWillMount() {
+      getBondModes().then(bondModes => this.setState({ bondModes }))
+    }
+
+    componentWillReceiveProps(props) {
+      this.forceUpdate()
+    }
+
+    _create = () => {
+      const { bondModes } = this.state
+      const { pool, state } = this.props
+      const {
+        bondedNetwork,
+        bondMode,
+        description,
+        mtu,
+        name,
+        pif,
+        pifs,
+        vlan,
+      } = state
+      return bondedNetwork
+        ? createBondedNetwork({
+            bondMode: bondMode || bondModes[0],
+            description,
+            mtu,
+            name,
+            pifs: map(pifs, pif => pif.id),
+            pool: pool.id,
+            vlan,
+          })
+        : createNetwork({
+            description,
+            mtu,
+            name,
+            pif: pif.id,
+            pool: pool.id,
+            vlan,
+          })
+    }
+
+    _selectPool = pool => {
+      const { pathname } = this.props.location
+      this.context.router.push({
+        pathname,
+        query: pool && { pool: pool.id },
+      })
+    }
+
+    _redirectOnSuccess = network =>
+      network !== undefined ? `pools/${this.props.pool.id}/network` : undefined
```

Comment:
What if `this.props.pool` is `undefined`?

Reply:
The form is displayed when the `pool !== undefined`